src/token/stream-parser.ts

```diff
@@ -48,68 +47,85 @@ class Parser extends Transform {
   suspended: boolean;
   next?: () => void;
 
-  constructor(debug: Debug, options: InternalConnectionOptions) {
-    super({ objectMode: true });
+  static async *parseTokens(iterable: AsyncIterable<Buffer>, debug: Debug, options: InternalConnectionOptions) {
```

Comment:
Would it make more sense to move this as it's own function and export this instead of the Parser class?

Reply:
Yes, I think that'd make sense. I skipped doing that, mostly because it will require us to update a bunch of `import` statements in multiple places and I wanted to save myself the headache of doing that. 😅 

Let's do that in a followup PR! 👍